runtime/doc/lua.txt

```diff
@@ -1035,10 +1035,24 @@ from within Lua.
         In Lua:
             `vim.opt.listchars = { space = '_', tab = '>~' }`
 
+    You can get an option's default value with >
+
+        vim.opt.wildignore:default()
+<
+    To set an option to its default value |:set-&| use: >
+
+        vim.opt.wildignore:reset()
+<
 
 In any of the above examples, to replicate the behavior |setlocal|, use
 `vim.opt_local`. Additionally, to replicate the behavior of |setglobal|, use
 `vim.opt_global`.
+
+To clear the local value of a |global-local| option set it to "nil". Example: >
+        vim.opt_local.makeprg = nil
+This is equivalent to >
+        :set makeprg<
```

Comment:
what does setting the value of a non `global-local` value to `nil` do? We should say what it's going to do in the docs (or whether it will error, etc.)

Reply:
Right now it [sets it to its default value](https://github.com/neovim/neovim/pull/15354#issuecomment-999102220). I'm not sure how I feel about that behavior though. It kind of makes sense, but could also be unexpected. The other option is making it a no-op.